src/metabase/models/card.clj

```diff
@@ -506,6 +507,10 @@
       (params/assert-valid-parameters changes)
       (params/assert-valid-parameter-mappings changes)
       (update-parameters-using-card-as-values-source changes)
+      ;; TODO: should be done in after-update
+      ;; has to place it here because changes is not available in after-update hook see toucan2#129
+      (when (contains? changes :dataset_query)
+        (query-analysis/analyze-async! id))
```

Comment:
Unfortunately the analysis requires that the database is updated first, moving this here could result in the analysis not actually being updated.

Reply:
we discussed this offline, and it turned out that even doing this after-update is not safe since the after-update is still under the transaction.

Chris said he'll re-visit the design later.

Reply:
Basically we relied on using the query value in the database in the analyzer, to ensure consistency (monotonically, eventually). If we just put the queries themselves on the queues, we can have race conditions even on single instance deployment where different version of the card with the `report_card` and `query_analysis` races.

For the near term though, I think that's just what we'll need to do. For "fast" updates we'll put the query itself on the async queue, and use it verbatim. For the sweeper though we'll still just put the card id on the sync queue, as it may we quite stale by the time we dispatch it. 

The longer term sweeper roadmap will audit the hash of each query against its analysis and we can get eventual consistency back that way, just without monotonicity.